azure/scope/machinepoolmachine.go

```diff
@@ -235,11 +236,13 @@ func (s *MachinePoolMachineScope) IsReady() bool {
 }
 
 // SetFailureMessage sets the AzureMachinePoolMachine status failure message.
+// TODO: should this be AzureMachinePoolMachine or AzureMachinePool?
```

Comment:
seems to me like it should be AzureMachinePoolMachine

Reply:
Sorry should have been more clear, I was saying the opposite: it should be setting the status of the AzureMachinePoolMachine, not the AzureMachinePool. I agree with you that the comment doesn't match the function, I'm saying the comment is right but the code is wrong (because this is in the machinepoolmachine.go file). @devigned might be able to confirm.

Reply:
I'm thinking this is probably copypasta that just happened to work since the scope contains a field that the MachinePoolScop contains.